src/daft-sql/src/planner.rs

```diff
@@ -686,7 +694,28 @@ impl SQLPlanner {
             SQLExpr::IsNotDistinctFrom(_, _) => {
                 unsupported_sql_err!("IS NOT DISTINCT FROM")
             }
-            SQLExpr::InList { .. } => unsupported_sql_err!("IN LIST"),
+            SQLExpr::InList {
+                expr,
+                list,
+                negated,
+            } => {
+                let expr = self.plan_expr(expr)?;
+                let list = list
+                    .iter()
+                    .map(|e| self.plan_lit(e))
+                    .collect::<SQLPlannerResult<Vec<_>>>()?;
```

Comment:
nit:
```suggestion
                    .try_collect()?;
```

and then let list: Vec<_> = ...

This is more appealing to me. Can be done with itertools or since we are using nightly it is a feature as well.